.github/workflows/flutter-test-linux-self-hosted.yml

```
@@ -0,0 +1,33 @@
+  push:
+    branches:
+      - '**'
+
+jobs:
+  test:
+    name: Flutter Test on Linux (self-hosted)
+    permissions:
+      contents: read
+    runs-on: self-hosted
+    steps:
+      - uses: actions/checkout@v3
```

actionlint (1.7.8)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)